src/toolkits/feature_engineering/word_trimmer.cpp

```diff
@@ -241,6 +243,9 @@ flexible_type word_trimmer_apply(const flexible_type& input,
                || run_mode == flex_type_enum::UNDEFINED
                || run_mode == flex_type_enum::DICT);
 
+  // do nothing; gigo
+  if (run_mode == flex_type_enum::UNDEFINED) return input;
```

Comment:
we should document the new behavior of the function when the input has `undefined` type.